eahub/base/views.py

```diff
@@ -94,28 +94,77 @@ def groups(request):
     )
 
 
-def get_groups_data():
-    rows = Group.objects.all()
+def get_groups_data(user, by_distance=False):
+    if hasattr(user, "profile") and by_distance:
+        if hasattr(user.profile, "lat"):
+            rows = get_groups_by_distance(user.profile)
+    else:
+        rows = Group.objects.all()
     map_data = [
         {
-            "lat": x.lat,
-            "lng": x.lon,
-            "label": x.name,
-            "active": x.is_active,
-            "path": f"/group/{x.slug}",
+            "lat": group.lat,
+            "lng": group.lon,
+            "label": group.name,
+            "active": group.is_active,
+            "path": f"/group/{group.slug}",
         }
-        for x in rows
-        if x.lat and x.lon
+        for group in rows
+        if group.lat and group.lon
     ]
     return {"rows": rows, "map_data": map_data}
 
 
+def get_groups_by_distance(profile):
+    # Difference in latitude (multiplied by 2 because latitude ranges only to 90)
```

Comment:
The correct way to calculate distances across latitude-and-longitude points is the haversine formula. Contrary to the below comment, you _can_ in fact implement this in a Django lookup; Django offers bindings to database math functions, including square-root and trigonometric functions.